app/custom_queries/hyrax/wings/custom_query_service.rb

```diff
@@ -0,0 +1,44 @@
+module Hyrax
+  module Wings
+    class CustomQueryService
+      # Defines custom queries for wings
+      # Use:
+      # => @custom_queries ||= Hyrax::Wings::CustomQueryService.new(query_service: ::Valkyrie.config.metadata_adapter.query_service)
+      # => resources = @custom_queries.find_many_by_alternate_ids(alternate_ids: ids, use_valkyrie: true)
+
+      def self.queries
+        [:find_many_by_alternate_ids]
+      end
+
+      attr_reader :query_service
+
+      def initialize(query_service:)
+        @query_service = query_service
+        @resource_factory = query_service.adapter.resource_factory
+      end
+
+      # implements a combination of two Valkyrie queries:
+      # => find_many_by_ids & find_by_alternate_identifier
+      # @param [Array<Valkyrie::ID, String>] ids
+      # @param [boolean] defaults to true; optionally return ActiveFedora::Base objects if false
+      # @return [Valkyrie::Resource or ActiveFedora::Base]
+      # NOTE: Ignores non-existent ids without raising an error
+      def find_many_by_alternate_ids(alternate_ids:, use_valkyrie: true)
```

Comment:
It seems like it would be good to implement this as an [`Enumerator`](https://ruby-doc.org/core-2.2.0/Enumerator.html), to avoid loading large collections into memory before iterating over them.

Reply:
I'm not sure how it avoids loading anything, but it does make the code easier to read.